test/sync/src/commonMain/kotlin/io/realm/test/mongodb/util/AdminApi.kt

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.test.mongodb.util
+
+import io.ktor.client.HttpClient
+import io.ktor.client.features.defaultRequest
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.delete
+import io.ktor.client.request.headers
+import io.ktor.client.request.post
+import io.ktor.client.request.request
+import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.readText
+import io.ktor.http.ContentType
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpMethod.Companion.Get
+import io.ktor.http.contentType
+import io.realm.internal.platform.runBlocking
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.serialization.InternalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import kotlinx.serialization.serializer
+
+private const val ADMIN_PATH = "/api/admin/v3.0"
+
+/**
+ * Wrapper around MongoDB Realm Server Admin functions needed for tests.
+ */
+interface AdminApi {
```

Comment:
Since this is just test code. Is there a good reason to keep the same interface/impl split? I don't have a strong opinion, just wondering

Reply:
It was just from the attempt of putting all the App related test methods on `TestApp` and provide easy access to the REST API while also separating the REST specific implementation into `AdminApi`. This way `TestApp` can implement the `AdminApi by AdminApiImp(...)` while the implementation is separate and we don't need to always define both a _test app_ and a _server admin_ instance (the latter is always needed to delete users on close, etc). Again I am up for counter proposals, but was just trying to address the `realm-java` annoyances of mixing test specific APIs on `App` and also always having to define an `admin` instance to access the REST api